pytorch_translate/dictionary.py

```diff
@@ -43,6 +46,71 @@ def __init__(
     def lexicon_indices_list(self):
         return list(self.lexicon_indices)
 
+    @classmethod
+    def create_vocab_file(
+        cls,
+        corpus_file: str,
+        vocab_file: str,
+        max_vocab_size: int,
+        tokens_with_penalty: Optional[str] = None,
+    ):
+        d = cls()
+        tokenizer.Tokenizer.add_file_to_dictionary(
+            filename=corpus_file, dict=d, tokenize=tokenizer.tokenize_line
+        )
+
+        # Set indices to receive penalty
+        if tokens_with_penalty:
+            # Assume input tokens are unique
+            lexicon = []
+            with open(tokens_with_penalty, "r", encoding="utf-8") as f:
+                for line in f:
+                    tokens = line.strip().split()
+                    if len(tokens) == 1:
+                        lexicon.append(tokens[0])
+
+            for token, token_index in d.indices.items():
+                if token in lexicon:
+                    d.lexicon_indices.add(token_index)
+
+        d.finalize()
+        d.save(vocab_file, threshold=0, nwords=max_vocab_size)
+        print(f"Generated new vocab file saved at {vocab_file}.")
```

Comment:
I think @liezl200 had looked at switching from print to Python logging, but decided it wasn't worth the trouble since the only thing it changed was that things were printed to stderr instead of stdout on the cluster?

I had also briefly considered logging before, but I think I ran into something where it seemed the default for the logging module was actually to not print anything at all unless you also set up stdout as the specified output or something? Plus I think some of our fairseq dependencies are still using print, and I wasn't sure what's a sensible interaction between `logging` and `tqdm` (that fairseq uses for progress bars) - so I ended up deciding to punt on that.

In principle I support the idea of using the logging module, but I think consistency is even more important. So, I'd be all for it if we decide to refactor everything to use `logging` - but until we have that, I'd probably just stick with using print.

Reply:
I agree with Wei, great explanation.

Reply:
Hi just in case it helps, you're right about logging not sending anything to stdout by default, but all it takes is one line to fix that (after which logging.info and logging.error and I can remember which others all get displayed as you'd expect) 
```
logging.basicConfig(format='%(asctime)s %(levelname)s %(message)s', level=logging.INFO , datefmt='%H:%M')
```